File: courseMaterial/code_section_4_topics/my_robot_tutorials/scripts/number_counter.py
#!/usr/bin/env python

import rospy
from std_msgs.msg import Int64

counter = 0
pub = None

def callback_number(msg):
	global counter
	counter += msg.data
	new_msg = Int64()
	new_msg.data = counter
	pub.publish(new_msg)

if __name__ == '__main__':
	rospy.init_node('number_counter')

	sub = rospy.Subscriber("/number", Int64, callback_number)

	pub = rospy.Publisher("/number_count", Int64, queue_size=10)

	rospy.spin()

File: courseMaterial/code_section_4_topics/my_robot_tutorials/src/number_counter.cpp
#include <ros/ros.h>
#include <std_msgs/Int64.h>

int counter = 0;
ros::Publisher pub;

void callback_number(const std_msgs::Int64& msg)
{
	counter += msg.data;
	std_msgs::Int64 new_msg;
	new_msg.data = counter;
	pub.publish(new_msg);
}

int main (int argc, char **argv)
{
	ros::init(argc, argv, "number_counter");
	ros::NodeHandle nh;

	ros::Subscriber sub = nh.subscribe("/number", 1000, callback_number);

	pub = nh.advertise<std_msgs::Int64>("/number_count", 10);

	ros::spin();
}
